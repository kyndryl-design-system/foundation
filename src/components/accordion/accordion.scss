/**
 * Copyright Kyndryl, Inc. 2023
 */
@use '../../scss/global.scss';
@use '../../scss/mixins/typography';

.kd-accordion {
  .accordion-item-container {
    border-radius: 4px;

    &.filled-headers {
      border: 1px solid var(--kd-color-border-light);
    }
  }

  .toggle-container {
    text-align: right;
    padding: 8px 0;
    @include typography.type-body-02;

    a:hover {
      text-decoration: underline;
    }
  }
}
