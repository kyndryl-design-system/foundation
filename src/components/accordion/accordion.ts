/**
 * Copyright Kyndryl, Inc. 2023
 */

import { html, LitElement } from 'lit';
import {
  property,
  state,
  customElement,
  queryAssignedElements,
} from 'lit/decorators.js';
import stylesheet from './accordion.scss';

/**
 * Accordion component.
 *
 * @slot unnamed - Holds Accordion Items (kd-accordion-item) that make up the accordion
 *
 */
@customElement('kd-accordion')
export class Accordion extends LitElement {
  static override styles = [stylesheet];

  /** Specifies whether to show numbers on the list items. */
  @property({ type: Boolean })
  showNumbers = false;

  /** Specifies the number to start at if the list has numbers. */
  @property({ type: Number })
  startNumber = 1;

  /** Specifies whether to show the accordion items with filled headers. */
  @property({ type: Boolean })
  filledHeaders = false;

  /** Display the accordion as compact or the default large size. */
  @property({ type: Boolean })
  compact = false;

  /** The string that displays on the toggle to expand all the accordion items. */
  @property({ type: String })
  expandLabel = 'Expand all items';

  /** The string that displays on the toggle to collapse all the accordion items. */
  @property({ type: String })
  collapseLabel = 'Collapse all items';

  /**
   * The state of the toggle controling the "expand all" functionality
   * @ignore
   */
  @state() private _allOpenState = false;

  /** Slotted children kd-accordion-item
   * @internal
   */
  @queryAssignedElements({ selector: 'kd-accordion-item' })
  _accordionItems!: Array<any>;

  protected _handleSlotChange() {
    this._updateChildren();
  }

  override willUpdate(changedProps: any) {
    if (
      changedProps.has('filledHeaders') ||
      changedProps.has('compact') ||
      changedProps.has('startNumber') ||
      changedProps.has('showNumbers')
    ) {
      this._updateChildren();
    }
  }

  protected _updateChildren() {
    this._accordionItems.forEach((item, index) => {
      item.setFilledHeader(this.filledHeaders);
      item.setCompact(this.compact);
      item.setIndex(this.startNumber + index);
      item.setShowNumbers(this.showNumbers);
    });
  }

  protected _openAllItems() {
    this._accordionItems.map((item) => {
      item.open();
    });
  }

  protected _closeAllItems() {
    this._accordionItems.map((item) => {
      item.close();
    });
  }

  private _toggleExpandAll() {
    if (this._allOpenState) {
      this._closeAllItems();
      this._allOpenState = false;
    } else {
      this._openAllItems();
      this._allOpenState = true;
    }
  }

  override render() {
    let itemContainerClasses = '';
    if (this.filledHeaders) itemContainerClasses += ' filled-headers';
    return html`
      <div class="kd-accordion">
        <div class="toggle-container">
          <a href="javascript:void(0)" @click="${this._toggleExpandAll}">
            ${this._allOpenState ? this.collapseLabel : this.expandLabel}
          </a>
        </div>

        <div class="accordion-item-container${itemContainerClasses}">
          <slot @slotchange="${this._handleSlotChange}"></slot>
        </div>
      </div>
    `;
  }
}
declare global {
  interface HTMLElementTagNameMap {
    'kd-accordion': Accordion;
  }
}
