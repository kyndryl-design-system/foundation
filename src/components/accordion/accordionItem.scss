/**
 * Copyright Kyndryl, Inc. 2023
 */
@use '../../scss/global';
@use '../../scss/mixins/typography';

:host {
  display: block;
}

.kd-accordion-item {
  padding: 0px 0px;
  display: block;
  background-color: var(--kd-color-background-page-default);
  border-radius: inherit;

  &.filled-header {
    .kd-accordion-item-title {
      background-color: var(--kd-color-background-accent-subtle);

      &:hover {
        outline-width: 1px;
        outline-color: var(--kd-color-border-ui-default);
      }
    }

    .kd-accordion-item-body {
      border-top: 1px solid var(--kd-color-border-variants-light);
    }
  }

  &.compact .kd-accordion-item-title {
    padding: 16px;
  }

  & .kd-accordion-item-body {
    padding: 24px 16px;
    display: none;
  }

  &.opened.filled-header .kd-accordion-item-title {
    border-radius: 0;
  }

  &.opened .kd-accordion-item-body {
    display: block;
  }

  .icon {
    vertical-align: middle;
  }
}

.expand-icon {
  display: flex;
  justify-content: center;
  align-content: center;
  flex-direction: column;
  margin-left: auto;
  transition: transform 150ms ease-out;

  .opened & {
    transform: rotate(-180deg);
  }
}

.kd-accordion-item-title {
  border-radius: inherit;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  background: none;
  position: relative;
  font-weight: var(--kd-font-weight-medium);
  padding: 24px 16px;
  outline: 2px solid transparent;
  transition: background-color 150ms ease-out, outline-color 150ms ease-out;
  outline-offset: -1px;

  &:hover {
    background-color: var(--kd-color-background-accent-subtle);
  }

  &:focus {
    outline-color: var(--kd-color-border-variants-focus);
  }

  .number,
  .icon {
    font-weight: var(--kd-font-weight-regular);
    padding-right: 2em;
    flex-grow: 0;
    flex-shrink: 1;
    flex-basis: auto;
    align-self: auto;
    order: 0;
    display: flex;
    justify-content: center;
    align-content: center;
    flex-direction: column;
  }

  .number {
    color: var(--kd-color-text-variant-placeholder);
  }

  .title {
    flex-grow: 1;
    flex-shrink: 1;
    flex-basis: auto;
    align-self: auto;
    order: 0;
  }

  .disabled & {
    color: var(--kd-color-text-link-level-disabled);
    background-color: var(--kd-color-background-page-default);
    outline: none;
  }
}

.kd-accordion-item-subtitle {
  @include typography.type-ui-02;
  font-weight: var(--kd-font-weight-regular);
  color: var(--kd-color-text-level-secondary);

  .disabled & {
    color: var(--kd-color-text-link-level-disabled);
  }
}

.disabled .kd-accordion-item-body {
  color: var(--kd-color-text-link-level-disabled);
}
