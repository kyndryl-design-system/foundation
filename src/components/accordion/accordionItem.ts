/**
 * Copyright Kyndryl, Inc. 2022
 */

import { html, LitElement } from 'lit';
import { state, property, customElement } from 'lit/decorators.js';
import { ifDefined } from 'lit/directives/if-defined.js';
import stylesheet from './accordionItem.scss';
import addIcon from '@carbon/icons/es/add/32';
import subtractIcon from '@carbon/icons/es/subtract/32';

/**
 * kd-accordion-item web component
 */
@customElement(`kd-accordion-item`)
export class AccordionItem extends LitElement {
  static override styles = [stylesheet];

  @property({ type: Boolean }) startOpened = false;

  @state() private _index = 1;
  @state() private _showNumber = false;
  @state() private _first = false;
  @state() private _opened = false;
  @state() private _filledHeader = false;
  @state() private _compact = false;

  override connectedCallback() {
    super.connectedCallback();
    if (this.startOpened) {
      this.open();
    }
  }

  setIndex(index: number) {
    this._index = index;
  }

  setFirst() {
    this._first = true;
  }

  setShowNumbers(value: boolean) {
    this._showNumber = value;
  }

  setFilledHeader(value: boolean) {
    this._filledHeader = value;
  }

  setCompact(value: boolean) {
    this._compact = value;
  }

  open() {
    if (!this._opened) this._toggleOpenState();
  }

  close() {
    if (this._opened) this._toggleOpenState();
  }

  private _handleClick(e: Event) {
    e.preventDefault();
    this._toggleOpenState();
  }

  private _handleKeypress(e: KeyboardEvent) {
    e.preventDefault();
    if (e.key == ' ' || e.key == 'Enter') this._toggleOpenState();
  }

  private _toggleOpenState() {
    if (this._opened) {
      this.ariaExpanded = 'false';
      this._opened = false;
    } else {
      this.ariaExpanded = 'true';
      this._opened = true;
    }
  }

  get numberTemplate() {
    if (this._showNumber) {
      return html`<div class="number">${this._index}</div>`;
    } else {
      return '';
    }
  }

  get iconTemplate() {
    if (this.querySelector('[slot="icon"]')) {
      return html`<div class="icon"><slot name="icon"></slot></div>`;
    } else {
      return '';
    }
  }

  get subtitleTemplate() {
    if (this.querySelector('[slot="subtitle"]')) {
      return html`
        <div class="kd-accordion-item-subtitle">
          <slot name="subtitle"></slot>
        </div>
      `;
    } else {
      return '';
    }
  }

  get expandIconTemplate() {
    if (this._opened)
      return html`
        <div class="expand-icon">
          <kd-icon .icon="${subtractIcon}"></kd-icon>
        </div>
      `;
    else {
      return html`
        <div class="expand-icon"><kd-icon .icon="${addIcon}"></kd-icon></div>
      `;
    }
  }

  override render() {
    let classAdditions = '';
    classAdditions += `kd-accordion-item`;

    if (this._first === true) {
      classAdditions += ` first-item`;
    }
    if (this._opened === true) {
      classAdditions += ` opened`;
    }
    if (this._filledHeader === true) {
      classAdditions += ` filled-header`;
    }
    if (this._compact === true) {
      classAdditions += ` compact`;
    }

    return html`
      <div
        class="${classAdditions}"
        aria-expanded="${ifDefined(
          this.ariaExpanded === null
            ? undefined
            : this.ariaExpanded === 'true'
            ? true
            : false
        )}"
        aria-controls="kd-accordion-item-detail-${this._index}"
        tabindex="${this._index}"
        @click="${(e: Event) => this._handleClick(e)}"
        @keypress="${(e: KeyboardEvent) => this._handleKeypress(e)}"
      >
        <div
          id="kd-accordion-item-title-${this._index}"
          class="kd-accordion-item-title"
        >
          ${this.iconTemplate} ${this.numberTemplate}
          <div class="title">
            <div><slot name="title"></slot></div>
            ${this.subtitleTemplate}
          </div>
          ${this.expandIconTemplate}
        </div>

        <div
          class="kd-accordion-item-body"
          id="kd-accordion-item-body-${this._index}"
          aria-labelledby="kd-accordion-item-title-${this._index}"
        >
          <div class="kd-accordion-item-detail">
            <slot name="body"></slot>
          </div>
        </div>
      </div>
    `;
  }
}
