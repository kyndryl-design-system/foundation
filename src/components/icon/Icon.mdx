import { Meta, Story, Canvas, Controls } from '@storybook/blocks';
import * as IconStories from './Icon.stories.js';
import { stringToReactHtml } from '../../common/helpers/helpers.ts';

import logoutIcon from '../../assets/svg/logout.svg';
import kyndrylIcon from '../../assets/svg/kyndryl-logo.svg';
import bridgeSmallIcon from '../../assets/svg/bridge-logo-small.svg';
import bridgeLargeIcon from '../../assets/svg/bridge-logo-large.svg';

<Meta of={IconStories} />

# Icons

<br />

## Carbon Icons

Most of our icon usage will come from IBM Carbon.

### Framework-specific Icons

It is recommended to use framework-specific versions of the Carbon components.

- [React](https://github.com/carbon-design-system/carbon/tree/main/packages/icons-react)
- [Svelte](https://github.com/carbon-design-system/carbon-icons-svelte)
- [Vue](https://github.com/carbon-design-system/carbon/tree/main/packages/icons-vue)
- [Angular](https://github.com/carbon-design-system/carbon-icons-angular)

### Vanilla Icons

Use the `<kd-icon>` component to make using [vanilla Carbon Icons](https://github.com/carbon-design-system/carbon/tree/main/packages/icons) (bundled with this library) easy. This is more intended for vanilla/Lit usage internally, but should work anywhere.

```js
import '@kyndryl-design-system/shidoka-foundation/components/icon';
import userAvatarIcon from '@carbon/icons/es/user--avatar/24';
```

```html
<kd-icon .icon="${userAvatarIcon}"></kd-icon>
```

<Canvas>
  <Story of={IconStories.Icon} />
</Canvas>

<Controls of={IconStories.Icon} />

### Gallery

[Carbon Icon Library](https://carbondesignsystem.com/guidelines/icons/library/)

<br />

## Kyndryl Icons

Kyndryl icons are custom SVG files that you can import directly from this library.

### Example Usage

It is recommended to use an SVG loader to inject the SVG as inline HTML into your app.

- Example Loaders:
  - [Rollup](https://www.npmjs.com/package/rollup-plugin-svg)
  - [Webpack](https://www.npmjs.com/package/svg-inline-loader)

SVG files can then be imported like this:

```js
import iconName from '@kyndryl-design-system/shidoka-foundation/assets/svg/<filename>.svg';
```

The method for placing them in your app at this point will vary based on the framework/loader.

<br />

### Gallery

<table>
  <thead>
    <tr>
      <th>Icon</th>
      <th>filename</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <span dangerouslySetInnerHTML={stringToReactHtml(logoutIcon)}></span>
      </td>
      <td>logout.svg</td>
    </tr>
    <tr>
      <td>
        <span dangerouslySetInnerHTML={stringToReactHtml(kyndrylIcon)}></span>
      </td>
      <td>kyndryl-logo.svg</td>
    </tr>
    <tr>
      <td>
        <span
          dangerouslySetInnerHTML={stringToReactHtml(bridgeSmallIcon)}
        ></span>
      </td>
      <td>bridge-logo-small.svg</td>
    </tr>
    <tr>
      <td>
        <span
          dangerouslySetInnerHTML={stringToReactHtml(bridgeLargeIcon)}
        ></span>
      </td>
      <td>bridge-logo-large.svg</td>
    </tr>
  </tbody>
</table>
