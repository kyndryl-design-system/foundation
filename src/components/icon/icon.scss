@import '../../common/scss/global.scss';

:host {
  display: inline-block;
}

svg {
  display: block;
}
