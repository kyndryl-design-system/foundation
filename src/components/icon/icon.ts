import { LitElement, html } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { unsafeHTML } from 'lit-html/directives/unsafe-html.js';
import { getAttributes, toString } from '@carbon/icon-helpers';
import IconScss from './icon.scss';

/**
 * Wrapper for vanilla Carbon icons to simplify usage.
 * @slot unnamed - The default slot for button content.
 */
@customElement('kyn-icon')
export class Icon extends LitElement {
  static override styles = IconScss;

  /** The imported Carbon icon. */
  @property({ type: Object })
  icon: any = {};

  /** Icon fill color. */
  @property({ type: String })
  fill = 'currentColor';

  /** Specify a size in pixels to override the imported Carbon icon's original size. */
  @property({ type: Number })
  sizeOverride = null;

  override render() {
    const attributes = JSON.parse(JSON.stringify(this.icon.attrs));
    attributes.fill = this.fill;

    if (this.sizeOverride) {
      attributes.width = this.sizeOverride;
      attributes.height = this.sizeOverride;
    }

    const iconString = toString({
      ...this.icon,
      attrs: getAttributes(attributes),
    });

    return html` ${unsafeHTML(iconString)} `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'kyn-icon': Icon;
  }
}
