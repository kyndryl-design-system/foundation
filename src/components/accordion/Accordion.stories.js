/**
 * Copyright Kyndryl, Inc. 2023
 */

import { html } from 'lit';

import './accordion';
import './accordionItem';
import '../icon';
import circleDashIcon from '@carbon/icons/es/circle-dash/24';
import checkmarkOutlineIcon from '@carbon/icons/es/checkmark--outline/24';
import errorFilledIcon from '@carbon/icons/es/error--filled/24';

export default {
  title: 'Components/Accordion',
  component: 'kd-accordion',
  subcomponents: { 'kd-accordion-item': 'kd-accordion-item' },
};

const args = {
  filledHeaders: false,
  compact: false,
  showNumbers: false,
  startNumber: 1,
  expandLabel: 'Expand all items',
  collapseLabel: 'Collapse all items',
};

export const Accordion = {
  args: args,
  render: (args) => {
    return html`
      <kd-accordion
        ?filledHeaders="${args.filledHeaders}"
        ?compact="${args.compact}"
        ?showNumbers="${args.showNumbers}"
        startNumber="${args.startNumber}"
        expandLabel="${args.expandLabel}"
        collapseLabel="${args.collapseLabel}"
      >
        <kd-accordion-item opened>
          <span slot="title">Accordion Title 1</span>
          <span slot="subtitle">Accordion subtitle 1</span>
          <div slot="body">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </div>
        </kd-accordion-item>

        <kd-accordion-item>
          <span slot="title">Accordion Title 2</span>
          <span slot="subtitle">Accordion subtitle 2</span>
          <div slot="body">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </div>
        </kd-accordion-item>

        <kd-accordion-item>
          <span slot="title">Accordion Title 3</span>
          <span slot="subtitle">Accordion subtitle 3</span>
          <div slot="body">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </div>
        </kd-accordion-item>
      </kd-accordion>
    `;
  },
};

Accordion.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/5TqtPa7KWfhJbQv6ELnbqf/Foundation?node-id=707%3A2396&mode=dev',
  },
};

export const AccordionWithIcons = {
  args: args,
  render: (args) => {
    return html`
      <kd-accordion
        ?filledHeaders="${args.filledHeaders}"
        ?compact="${args.compact}"
        ?showNumbers="${args.showNumbers}"
        startNumber="${args.startNumber}"
        expandLabel="${args.expandLabel}"
        collapseLabel="${args.collapseLabel}"
      >
        <kd-accordion-item opened>
          <kd-icon
            slot="icon"
            .icon=${circleDashIcon}
            fill="var(--kd-color-text-placeholder)"
          ></kd-icon>
          <span slot="title">Accordion Title 1</span>
          <span slot="subtitle">Accordion subtitle 1</span>
          <div slot="body">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </div>
        </kd-accordion-item>

        <kd-accordion-item>
          <kd-icon
            slot="icon"
            .icon=${checkmarkOutlineIcon}
            fill="var(--kd-color-text-success)"
          ></kd-icon>
          <span slot="title">Accordion Title 2</span>
          <span slot="subtitle">Accordion subtitle 2</span>
          <div slot="body">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </div>
        </kd-accordion-item>

        <kd-accordion-item>
          <kd-icon
            slot="icon"
            .icon=${errorFilledIcon}
            fill="var(--kd-color-text-destructive)"
          ></kd-icon>
          <span slot="title">Accordion Title 3</span>
          <span slot="subtitle">Accordion subtitle 3</span>
          <div slot="body">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </div>
        </kd-accordion-item>
      </kd-accordion>
    `;
  },
};
