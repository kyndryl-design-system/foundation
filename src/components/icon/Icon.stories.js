import { html } from 'lit';
import './icon';
import userIcon from '@carbon/icons/es/user--avatar/24';

export default {
  title: 'Components/Icon',
  component: 'kd-icon',
  argTypes: {
    fill: {
      control: { type: 'color' },
    },
    sizeOverride: {
      control: { type: 'number', min: 10, max: 100 },
    },
  },
};

export const Icon = {
  args: {
    icon: userIcon,
    fill: 'currentColor',
    sizeOverride: undefined,
  },
  render: (args) => html`
    <kd-icon
      .icon=${args.icon}
      fill=${args.fill}
      sizeOverride=${args.sizeOverride}
    ></kd-icon>
  `,
};
