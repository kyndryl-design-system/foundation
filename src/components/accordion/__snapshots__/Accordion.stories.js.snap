// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Accordion Accordion smoke-test 1`] = `
<kd-accordion startnumber="1"
              expandlabel="Expand all items"
              collapselabel="Collapse all items"
>
  <kd-accordion-item opened>
    <span slot="title">
      Accordion Title 1
    </span>
    <span slot="subtitle">
      Accordion subtitle 1
    </span>
    <div slot="body">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </div>
  </kd-accordion-item>
  <kd-accordion-item>
    <span slot="title">
      Accordion Title 2
    </span>
    <span slot="subtitle">
      Accordion subtitle 2
    </span>
    <div slot="body">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </div>
  </kd-accordion-item>
  <kd-accordion-item>
    <span slot="title">
      Accordion Title 3
    </span>
    <span slot="subtitle">
      Accordion subtitle 3
    </span>
    <div slot="body">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </div>
  </kd-accordion-item>
</kd-accordion>
<div class="kd-accordion">
  <div class="toggle-container">
    <a href="javascript:void(0)">
      Expand all items
    </a>
  </div>
  <div class="accordion-item-container">
    <slot>
    </slot>
  </div>
</div>
<div class=" kd-accordion-item opened ">
  <div class="kd-accordion-item-title"
       tabindex="0"
       role="button"
       aria-controls="kd-accordion-item-body-1"
       aria-expanded="true"
       id="kd-accordion-item-title-1"
  >
    <div class="title">
      <slot name="title">
      </slot>
      <div class="kd-accordion-item-subtitle">
        <slot name="subtitle">
        </slot>
      </div>
    </div>
    <div class="expand-icon">
      <kd-icon>
      </kd-icon>
    </div>
  </div>
  <div class="kd-accordion-item-body"
       role="region"
       id="kd-accordion-item-body-1"
       aria-labelledby="kd-accordion-item-title-1"
  >
    <div class="kd-accordion-item-detail">
      <slot name="body">
      </slot>
    </div>
  </div>
</div>
<div class=" kd-accordion-item ">
  <div class="kd-accordion-item-title"
       tabindex="0"
       role="button"
       aria-controls="kd-accordion-item-body-2"
       aria-expanded="false"
       id="kd-accordion-item-title-2"
  >
    <div class="title">
      <slot name="title">
      </slot>
      <div class="kd-accordion-item-subtitle">
        <slot name="subtitle">
        </slot>
      </div>
    </div>
    <div class="expand-icon">
      <kd-icon>
      </kd-icon>
    </div>
  </div>
  <div class="kd-accordion-item-body"
       role="region"
       id="kd-accordion-item-body-2"
       aria-labelledby="kd-accordion-item-title-2"
  >
    <div class="kd-accordion-item-detail">
      <slot name="body">
      </slot>
    </div>
  </div>
</div>
<div class=" kd-accordion-item ">
  <div class="kd-accordion-item-title"
       tabindex="0"
       role="button"
       aria-controls="kd-accordion-item-body-3"
       aria-expanded="false"
       id="kd-accordion-item-title-3"
  >
    <div class="title">
      <slot name="title">
      </slot>
      <div class="kd-accordion-item-subtitle">
        <slot name="subtitle">
        </slot>
      </div>
    </div>
    <div class="expand-icon">
      <kd-icon>
      </kd-icon>
    </div>
  </div>
  <div class="kd-accordion-item-body"
       role="region"
       id="kd-accordion-item-body-3"
       aria-labelledby="kd-accordion-item-title-3"
  >
    <div class="kd-accordion-item-detail">
      <slot name="body">
      </slot>
    </div>
  </div>
</div>
`;

exports[`Components/Accordion AccordionWithIcons smoke-test 1`] = `
<kd-accordion startnumber="1"
              expandlabel="Expand all items"
              collapselabel="Collapse all items"
>
  <kd-accordion-item opened>
    <kd-icon slot="icon"
             fill="var(--kd-color-text-placeholder)"
    >
    </kd-icon>
    <span slot="title">
      Accordion Title 1
    </span>
    <span slot="subtitle">
      Accordion subtitle 1
    </span>
    <div slot="body">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </div>
  </kd-accordion-item>
  <kd-accordion-item>
    <kd-icon slot="icon"
             fill="var(--kd-color-text-success)"
    >
    </kd-icon>
    <span slot="title">
      Accordion Title 2
    </span>
    <span slot="subtitle">
      Accordion subtitle 2
    </span>
    <div slot="body">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </div>
  </kd-accordion-item>
  <kd-accordion-item>
    <kd-icon slot="icon"
             fill="var(--kd-color-text-destructive)"
    >
    </kd-icon>
    <span slot="title">
      Accordion Title 3
    </span>
    <span slot="subtitle">
      Accordion subtitle 3
    </span>
    <div slot="body">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </div>
  </kd-accordion-item>
</kd-accordion>
<div class="kd-accordion">
  <div class="toggle-container">
    <a href="javascript:void(0)">
      Expand all items
    </a>
  </div>
  <div class="accordion-item-container">
    <slot>
    </slot>
  </div>
</div>
<div class=" kd-accordion-item opened ">
  <div class="kd-accordion-item-title"
       tabindex="0"
       role="button"
       aria-controls="kd-accordion-item-body-1"
       aria-expanded="true"
       id="kd-accordion-item-title-1"
  >
    <div class="icon">
      <slot name="icon">
      </slot>
    </div>
    <div class="title">
      <slot name="title">
      </slot>
      <div class="kd-accordion-item-subtitle">
        <slot name="subtitle">
        </slot>
      </div>
    </div>
    <div class="expand-icon">
      <kd-icon>
      </kd-icon>
    </div>
  </div>
  <div class="kd-accordion-item-body"
       role="region"
       id="kd-accordion-item-body-1"
       aria-labelledby="kd-accordion-item-title-1"
  >
    <div class="kd-accordion-item-detail">
      <slot name="body">
      </slot>
    </div>
  </div>
</div>
<svg focusable="false"
     preserveaspectratio="xMidYMid meet"
     xmlns="http://www.w3.org/2000/svg"
     fill="var(--kd-color-text-placeholder)"
     aria-hidden="true"
     width="24"
     height="24"
     viewbox="0 0 32 32"
>
  <path d="M7.7 4.7a14.7 14.7 0 00-3 3.1L6.3 9A13.26 13.26 0 018.9 6.3zM4.6 12.3l-1.9-.6A12.51 12.51 0 002 16H4A11.48 11.48 0 014.6 12.3zM2.7 20.4a14.4 14.4 0 002 3.9l1.6-1.2a12.89 12.89 0 01-1.7-3.3zM7.8 27.3a14.4 14.4 0 003.9 2l.6-1.9A12.89 12.89 0 019 25.7zM11.7 2.7l.6 1.9A11.48 11.48 0 0116 4V2A12.51 12.51 0 0011.7 2.7zM24.2 27.3a15.18 15.18 0 003.1-3.1L25.7 23A11.53 11.53 0 0123 25.7zM27.4 19.7l1.9.6A15.47 15.47 0 0030 16H28A11.48 11.48 0 0127.4 19.7zM29.2 11.6a14.4 14.4 0 00-2-3.9L25.6 8.9a12.89 12.89 0 011.7 3.3zM24.1 4.6a14.4 14.4 0 00-3.9-2l-.6 1.9a12.89 12.89 0 013.3 1.7zM20.3 29.3l-.6-1.9A11.48 11.48 0 0116 28v2A21.42 21.42 0 0020.3 29.3z">
  </path>
</svg>
<div class=" kd-accordion-item ">
  <div class="kd-accordion-item-title"
       tabindex="0"
       role="button"
       aria-controls="kd-accordion-item-body-2"
       aria-expanded="false"
       id="kd-accordion-item-title-2"
  >
    <div class="icon">
      <slot name="icon">
      </slot>
    </div>
    <div class="title">
      <slot name="title">
      </slot>
      <div class="kd-accordion-item-subtitle">
        <slot name="subtitle">
        </slot>
      </div>
    </div>
    <div class="expand-icon">
      <kd-icon>
      </kd-icon>
    </div>
  </div>
  <div class="kd-accordion-item-body"
       role="region"
       id="kd-accordion-item-body-2"
       aria-labelledby="kd-accordion-item-title-2"
  >
    <div class="kd-accordion-item-detail">
      <slot name="body">
      </slot>
    </div>
  </div>
</div>
<svg focusable="false"
     preserveaspectratio="xMidYMid meet"
     xmlns="http://www.w3.org/2000/svg"
     fill="var(--kd-color-text-success)"
     aria-hidden="true"
     width="24"
     height="24"
     viewbox="0 0 32 32"
>
  <path d="M14 21.414L9 16.413 10.413 15 14 18.586 21.585 11 23 12.415 14 21.414z">
  </path>
  <path d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,26A12,12,0,1,1,28,16,12,12,0,0,1,16,28Z">
  </path>
</svg>
<div class=" kd-accordion-item ">
  <div class="kd-accordion-item-title"
       tabindex="0"
       role="button"
       aria-controls="kd-accordion-item-body-3"
       aria-expanded="false"
       id="kd-accordion-item-title-3"
  >
    <div class="icon">
      <slot name="icon">
      </slot>
    </div>
    <div class="title">
      <slot name="title">
      </slot>
      <div class="kd-accordion-item-subtitle">
        <slot name="subtitle">
        </slot>
      </div>
    </div>
    <div class="expand-icon">
      <kd-icon>
      </kd-icon>
    </div>
  </div>
  <div class="kd-accordion-item-body"
       role="region"
       id="kd-accordion-item-body-3"
       aria-labelledby="kd-accordion-item-title-3"
  >
    <div class="kd-accordion-item-detail">
      <slot name="body">
      </slot>
    </div>
  </div>
</div>
<svg focusable="false"
     preserveaspectratio="xMidYMid meet"
     xmlns="http://www.w3.org/2000/svg"
     fill="var(--kd-color-text-destructive)"
     aria-hidden="true"
     width="24"
     height="24"
     viewbox="0 0 24 24"
>
  <path d="M12,1C5.9,1,1,5.9,1,12s4.9,11,11,11s11-4.9,11-11S18.1,1,12,1z M16.3,17.5L6.5,7.7l1.2-1.2l9.8,9.8L16.3,17.5z">
  </path>
  <path fill="none"
        d="M16.3,17.5L6.5,7.7l1.2-1.2l9.8,9.8L16.3,17.5z"
        data-icon-path="inner-path"
        opacity="0"
  >
  </path>
</svg>
`;
